src/App.tsx

```diff
@@ -1,10 +1,55 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { ErrorMessage } from './components/ErrorMessage/ErrorMessage';
+import { TodoList } from './components/TodoList/TodoList';
+import { Todo } from './types/Todo';
+import { getTodos } from './api/todos';
+import { TodoFilter } from './components/TodoFilter/TodoFilter';
+import { FilterBy } from './utils/enums';
 
-const USER_ID = 0;
+const USER_ID = 10897;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [hasActive, setHasActive] = useState(false);
+  const [hasCompleted, setHasCompleted] = useState(false);
+  const [errorText, setErrorText] = useState<null | string>(null);
+  const [filterBy, setFilterBy] = useState(FilterBy.ALL);
+
+  const resetError = () => {
+    setErrorText(null);
+  };
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(response => {
+        setTodos(response);
+        setHasActive(response.some(todo => !todo.completed));
+        setHasCompleted(response.some(todo => todo.completed));
+      })
+      .catch((error) => {
+        setErrorText(error.message);
+        setTimeout(resetError, 3000);
+      });
+  }, []);
+
+  const handleFilterChange = useCallback((event: React.MouseEvent) => {
+    const target = event.target as HTMLButtonElement;
+
+    setFilterBy(target.innerText.toLowerCase() as FilterBy);
```

Comment:
```suggestion
    setFilterBy(event.target.innerText.toLowerCase() as FilterBy);
```

Reply:
@silchyn I would like to do that, but TS doesn't like that "Property 'innerText' does not exist on type 'EventTarget'"